webapp/src/components/Modal/BidConfirmationModal/BidConfirmationModal.js

```diff
@@ -24,6 +32,60 @@ export default class BidConfirmationModal extends React.PureComponent {
     onAuthorize: PropTypes.func
   }
 
+  constructor(props) {
+    super(props)
+    this.state = {
+      isLoading: false,
+      balance: null
+    }
+  }
+
+  componentWillMount() {
+    const { token, address } = this.props
+    this.fetchBalance(token, address)
+  }
+
+  componentWillReceiveProps({ token, address }) {
+    if (this.props.token !== token || this.props.address !== address) {
+      this.fetchBalance(token, address)
+    }
+  }
+
+  fetchBalance = async (symbol, address) => {
+    this.setState({ balance: null, isLoading: true })
+
+    try {
+      const contractAddress = TOKEN_ADDRESSES[symbol]
+      const response = await axios.get(
+        `https://api.etherscan.io/api?module=account&action=tokenbalance&contractaddress=${contractAddress}&address=${address}&tag=latest&apikey=${env.get(
+          'REACT_APP_ETHERSCAN_API_KEY'
+        )}`
+      )
+      const balance =
+        Number(response.data.result) / (symbol === 'ZIL' ? 10 ** 12 : 10 ** 18)
+      if (isNaN(balance)) {
+        throw new Error('Error fetching balance')
+      }
+      this.setState({
+        balance,
+        isLoading: false
+      })
+    } catch (error) {
+      window.Rollbar.info('Fetching balance via Infura')
+      const contractName = token.getContractNameBySymbol(symbol)
+      const contract = eth.getContract(contractName)
+      const balance = await contract.balanceOf(address)
+
+      this.setState({
+        balance:
+          balance && typeof balance.toNumber === 'function'
+            ? balance.toNumber()
+            : balance,
+        isLoading: false
+      })
```

Comment:
should we worry about this failing too? (returning 0) I don't know how ofter etherscan fails of if it has a rate limit, asking just to be safe because we're disabling the button

Reply:
i didn't get any errors fetching from etherscan, i left there just in case, but we can remove it all together if you think it makes it messier.